app/src/main/java/net/gsantner/markor/activity/SettingsActivity.java

```diff
@@ -281,6 +287,24 @@ public void onFsViewerConfig(FilesystemViewerData.Options dopt) {
                     }
                     return true;
                 }
+                case R.string.pref_key__default_encryption_password: {
```

Comment:
Please use EditTextPreference. Visibility of the preference can be set in the doUpdatePreferences method of SettingsActivity  (see the setVisible calls)

You can clear the preference value after it gets set and just keep value in passwordstore.

Reply:
Can you go more into detail? I'm not an android-expert and didn't understand what I should do.

Reply:
Use a EditTextPreference in xml, like this one

![grafik](https://user-images.githubusercontent.com/6735650/78479463-c8421480-7734-11ea-82ef-39d416f5949a.png)

And then in the SettingsActivity you get a callback with what the user entered:

![grafik](https://user-images.githubusercontent.com/6735650/78480039-f0317800-7734-11ea-97e3-b135eaa8abb2.png)

You can then clear this what the user entered (`set(key, "")`) and move the password over to your own password store.

Reply:
Will the text input have the property textVisiblePassword? Otherwise the password will be stored by the keyboard.

Reply:
Great! I hope I got it now, the last commit.